plugins/pingme.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+#  pingme.rb
+#  Author: Ckat
+#  ------------------------
+#  A canonical Cinch plugin for yossarian-bot.
+#  combot's pingme command, lazily implemented
+#  ------------------------
+#  This code is licensed by William Woodruff under the MIT License.
+#  http://opensource.org/licenses/MIT
+
+require_relative "yossarian_plugin"
+
+class PingMe < YossarianPlugin
+  include Cinch::Plugin
+  use_blacklist
+
+  def initialize(*args)
+    super
+    @nick = ""
+    @channel = ""
+    @timestamp = {}
+    @sent = false
+  end
+
+  def usage
+    "!pingme - Get pinged"
+  end
+
+  def match?(cmd)
+    cmd =~ /^(!)?pingme$/
+  end
+
+  match /pingme$/, method: :ctcp_ping_req
+  listen_to :ctcp, method: :ctcp_ping_recv
+
+  def format_ping_time
```

Comment:
There should be either a helper in the stdlib *or* a popular library to do these kinds of time diffs. I'd be okay with either instead of embedding this utility function. We might even use one elsewhere in the codebase, although I could be misremembering.

Reply:
yup there is one in bot_info.rb, missed that https://github.com/woodruffw/yossarian-bot/blob/9c8ee858c02ac3e9d2f5594c35f53354d8fcaa70/plugins/bot_info.rb#L41-L42

Reply:
it does, however, only go to seconds, not milliseconds